extensions/amp-analytics/0.1/requests.js

```diff
@@ -207,18 +240,38 @@ export class RequestHandler {
    */
   fire_() {
     const {
+      requestOriginPromise_: requestOriginPromise,
       baseUrlTemplatePromise_: baseUrlTemplatePromise,
       baseUrlPromise_: baseUrlPromise,
       batchSegmentPromises_: segmentPromises,
     } = this;
     const trigger = /** @type {!JsonObject} */ (this.lastTrigger_);
     this.reset_();
 
-    baseUrlTemplatePromise.then(preUrl => {
+    // preconnect to requestOrigin if available, otherwise baseUrlTemplate
+    const preconnectPromise = requestOriginPromise
+      ? requestOriginPromise
+      : baseUrlTemplatePromise;
+
+    preconnectPromise.then(preUrl => {
```

Comment:
nit: preconnect happens in parallel. What about we change the nested promise chain to two promises. 
```
preconnectPromise.then(preUrl => {
  preconnect()
});
//then
Promise.all([requestOrignPromise, baseUrlPromise....]).then(() => {
});
```

It's an optimization. Feel free to address this later or let me know if there's different idea.